### DISCUSSION TITLE
Include metadata of getItems query

### DISCUSSION THREAD

**Discussion by kieusonlam** [Intevel/nuxt-directus#26] [Q&A] [answered]
Hi,

I'm not sure what is the best way to implement this but I think meta data is needed for paging when quering a set of data.

How did you guys handle this right now?

**Comment by Intevel** [comment]
Currently we have the page param which you can provide. What do you mean with "meta data"

https://docs.directus.io/reference/query/#page

**Reply by kieusonlam** [reply]
Yes but it's not a part of params

https://docs.directus.io/reference/query/#metadata

```ts
export interface DirectusQueryParams {
  fields?: Array<string>;
  sort?: string | Array<string>;
  filter?: Record<string, unknown>;
  limit?: number;
  offset?: number;
  page?: number;
  alias?: string | Array<string>;
  deep?: Record<string, unknown>;
}
```
https://github.com/Intevel/nuxt-directus/blob/7919b63a06d0b7122c5a4fec15ce5115d6d82c27/src/runtime/composables/useDirectusItems.ts#L23

Also the return data is in the diferrent path

![image](https://user-images.githubusercontent.com/13093313/165131098-17af199c-f2c6-42eb-8864-ef928d7a8a96.png)

**Comment by Intevel** [comment]
So the `page` param isn't part of the meta data. You can provide it with normal params. But you are right currently the https://docs.directus.io/reference/query/#metadata is not supported. I will make an issue for that.

**Comment by leosin** [comment]
Is there any way to do pagination based on this module?
The official documentation is also only through`?meta=total_count` to get the total quantity

**Reply by Intevel** [reply]
@leosin I just pushed a new version, with the `meta` params support 👯

**Reply by leosin** [reply]
awesome!!! 😊 👍👍👍

**Comment by Intevel** [comment, marked as answer]
@kieusonlam Can you mark this as answered please?

**Comment by leosin** [comment]
Sorry, I have a stupid question
After version 2.4, the meta parameter was added, but I didn't have meta when I got the result, but I could see it when I requested it.

<img width="413" alt="image" src="https://user-images.githubusercontent.com/3676670/182231857-2ca5df73-cb26-4feb-a86b-c83497f03fd2.png">
<img width="565" alt="image" src="https://user-images.githubusercontent.com/3676670/182232046-ee8860dd-8087-4338-88b8-d6464cfdde67.png">

```ts
const { getItems } = useDirectusItems()
const fetchResults = async (i) => {
  try {
    const results = await getItems({
      collection: 'fagui',
      params: {
        search: i,
        limit: 10,
        page: 1,
        meta: 'filter_count',
      },
    })
    console.log('Results:', results)
  }
  catch (e) {}
   @@@}
await fetchResults(keyword.value)
```
Am I missing something?
@Intevel

**Reply by Intevel** [reply]
What do you mean with "but I didn't have meta when I got the result" 

Youre filter_count meta is 9, and you receive 9 elements. I don't understand your problem.

**Comment by leosin** [comment]
sorry , It may be that I did not express it clearly

Although the meta parameter can be used.

but `getItems()`  only get the value in 'data'
```ts
data:[.....]
````
 how can i get `filter_count` value
@Intevel 

```ts
const fetchPosts = async (i) => {
  const { data: blog }: any = await useFetch('/items/fagui/', {
    baseURL: useRuntimeConfig().public.baseURL,
    headers: { Authorization: `Bearer ${useRuntimeConfig().public.token}` },
    params: {
      search: i as string,
      limit: 10 as number, // maximum returned
      page: 1,
      meta: 'filter_count',
    },
  })
  // the maximum number of items that will be returned
  console.log('limit:', blog.value.data.length)

  // like this ,i can get meta.filter_count
  console.log('total:', blog.value.meta.filter_count)
}
fetchPosts(keyword)

**Reply by leosin** [reply]
<img width="633" alt="image" src="https://user-images.githubusercontent.com/3676670/182366008-0664f10a-8ea7-4a06-82ae-9294e808df4b.png">

**Reply by albericobenhur** [reply]
I tried to use the filter parameter and it gave undefined

![Captura de Tela 2022-08-24 às 17 29 05](https://user-images.githubusercontent.com/86002589/186517789-b446d8c1-f31f-4b37-adbc-b299cc11c454.png)

![Captura de Tela 2022-08-24 às 17 34 09](https://user-images.githubusercontent.com/86002589/186518089-e00be610-c212-460b-8dc2-0724e14e2185.png)

**Reply by Intevel** [reply]
Hey @leosin, in the newest release you can access meta. 
`getItems` will now return data and meta

![image](https://user-images.githubusercontent.com/59223342/187069655-8f370250-841b-41bb-a691-686e7bb1e064.png)

**Reply by leosin** [reply]
that's great job, thank you very much  💯 💗 👍
